src/dmd/statementsem.d

```diff
@@ -1256,6 +1261,11 @@ private extern (C++) final class StatementSemanticVisitor : Visitor
                             }
                             fs.key.range = new IntRange(SignExtendedNumber(0), dimrange.imax);
                         }
+                        else if (!Type.tsize_t.implicitConvTo(var.type))
+                        {
+                            fs.deprecation("foreach: loop index implicitly converted from `size_t` to `%s`",
+                                           fs.key.type.toChars());
```

Comment:
Notice a minor thing. I would have been nice to have the location point to the loop index variable instead of the `foreach` statement.